src/geo/crs/CRS.Earth.js

```diff
@@ -24,9 +24,11 @@ export var Earth = Util.extend({}, CRS, {
 		var rad = Math.PI / 180,
 		    lat1 = latlng1.lat * rad,
 		    lat2 = latlng2.lat * rad,
-		    a = Math.sin(lat1) * Math.sin(lat2) +
-		        Math.cos(lat1) * Math.cos(lat2) * Math.cos((latlng2.lng - latlng1.lng) * rad);
-
-		return this.R * Math.acos(Math.min(a, 1));
+		    dLat = (latlng2.lat - latlng1.lat) * rad,
+		    dLon = (latlng2.lng - latlng1.lng) * rad,
+		    a = Math.sin(dLat / 2) * Math.sin(dLat / 2) +
```

Comment:
Please store `Math.sin(dLat / 2)` in a separate variable. The current code will make two calls to `Math.sin()`, and I do not expect the JITs to magically optimize this. 

Same goes for `Math.sin(dLon / 2)`.